cmd/generic-handlers.go

```diff
@@ -729,6 +727,10 @@ func (f bucketForwardingHandler) ServeHTTP(w http.ResponseWriter, r *http.Reques
 // on a bucket to the right bucket location, bucket to IP configuration
 // is obtained from centralized etcd configuration service.
 func setBucketForwardingHandler(h http.Handler) http.Handler {
+	if globalDNSConfig == nil || len(globalDomainNames) == 0 || !globalBucketFederation {
```

Comment:
These are set when reading the config `initConfig(objAPI ObjectLayer)`, but I heavily suspect that the server handlers are set up before that.

So AFAICT one or more of these will always fail and it will never get added.